recipe/meta.yaml

```diff
@@ -86,7 +90,7 @@ requirements:
     - proj
     - udunits2
     - zlib
-    - esmf <8.6.0
+    - esmf * mpi_*
```

Comment:
If I understand it correctly, loading a grid from file is done using [this function](https://github.com/esmf-org/esmf/blob/e5e125f35496de8e05121ec90eb140189777fca5/src/Infrastructure/Mesh/src/ESMCI_Mesh_FileIO.C#L122-L135) which requires parallelio which is only available in the [mpi builds](https://github.com/conda-forge/esmf-feedstock/blob/e38a02a55689126f5dd70805389f0c202d471af6/recipe/meta.yaml#L91).